action/lib/parse.js

```diff
@@ -24,18 +50,77 @@ export default function (title, target) {
     return process.exit(0) // soft exit
   }
 
+  let isDev = devDependencyRegEx.test(title);
+  let isProd = dependencyRegEx.test(title);
+  const isSecurity = securityRegEx.test(title) || labels.includes("security") || labels.includes("Security");
+
+  if (!isDev && !isProd) {
+    // couldn't extract the dependency type from the title, try to read package.json
+    try {
+      const packageJsonPath = path.join(ghWorkspace, 'package.json')
+      const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, "utf8"));
+      isDev = !!packageJson.devDependencies && depName in packageJson.devDependencies
+      isProd = !!packageJson.dependencies && depName in packageJson.dependencies
+    } catch (e) {
+      console.dir(e);
+     }
+  }
+  if (!isDev && !isProd) {
+    // we failed again, assume its a production dependency
+    core.info(`failed to parse dependency type, assuming it is a production dependency`)
+    isProd = true
+  }
+
   // log
   core.info(`from: ${from.version}`)
   core.info(`to: ${to.version}`)
+  core.info(`dependency type: ${isDev ? "development" : isProd ? "production" : "unknown"}`)
+  core.info(`security critical: ${isSecurity}`)
+
+  // convert target to the automerged_updates syntax
+  const configPath = path.join(ghWorkspace, '.github/auto-merge.yml')
```

Comment:
I typically avoid using `/` for path separation and simply use `.join` to take care of it to be platform agnostic ...

Reply:
todo: I would move all of the yaml file reading out of `parse.js` and into separate step (perhaps in `index.js`) so that `parse.js` just get passed the final config.

Reply:
> I typically avoid using `/` for path separation and simply use `.join` to take care of it to be platform agnostic ...

In my experience, `path.join` does OS agnostic normalization:
![grafik](https://user-images.githubusercontent.com/17641229/92606877-bf410a80-f2b3-11ea-90b3-2cbf9af478cd.png)

I can split it into 3 arguments though.

Reply:
oh nice! I did not realize that ... good to know.